apis/concierge/authentication/v1alpha1/types_tls.go.tmpl

```diff
@@ -1,11 +1,30 @@
-// Copyright 2020-2022 the Pinniped contributors. All Rights Reserved.
+// Copyright 2020-2024 the Pinniped contributors. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
 package v1alpha1
 
+// CABundleSource provides a source for CA bundle used for client-side TLS verification.
+type CABundleSource struct {
+	// Whether the CA bundle is being sourced from a kubernetes secret or a configmap.
+	// Secrets must be of type kubernetes.io/tls or Opaque.
+	// For configmaps, the value associated with the key is not expected to be base64 encoded.
+	// +kubebuilder:validation:Enum=Secret;ConfigMap
+	Kind string `json:"kind"`
+	// Name of the secret or configmap from which to read the CA bundle.
+	// The referenced secret or configmap must be created in the same namespace where Pinniped Concierge is installed.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Key within the secret or configmap from which to read the CA bundle.
```

Comment:
Perhaps we should also document that the value at this key must not be empty, and must be a PEM-encoded certificate bundle.

Reply:
will be done in next push